File: web3rpc/rpc-specs/code-samples/javascript/debug/vMTracing/traceChain.js
const { JsonRpcProvider } = require("@kaiachain/ethers-ext");

(async () => {
    const provider = new JsonRpcProvider("https://public-en-kairos.node.kaia.io");

    const startBlock = "1000";
    const endBlock = "1001";
    const options = { tracer: "revertTracer" };
    // The full list of JSON-RPC is available at:
    // https://docs.kaia.io/references/json-rpc/references
    provider.debug.traceChain(
        startBlock,
        endBlock,
        options,
        null,
        (err, data, response) => {

            console.log("Trace chain", data);
        }
    );
})();

File: web3rpc/rpc-specs/code-samples/python/debug/vMTracing/traceChain.py
from web3 import Web3
from web3py_ext import extend

host = "https://public-en-kairos.node.kaia.io"

startBlock = "10000"
endBlock = "10001"
options = {"tracer":"revertTracer"}
w3 = Web3(Web3.HTTPProvider(host))
debug_response = w3.debug.trace_call(startBlock, endBlock,options)

print(debug_response)
